src/rgw/rgw_swift_auth.cc

```diff
@@ -437,8 +436,7 @@ static int build_token(const string& swift_user,
 
   bufferptr p(CEPH_CRYPTO_HMACSHA1_DIGESTSIZE);
 
-  char buf[bl.length() * 2 + 1];
-  buf_to_hex((const unsigned char *)bl.c_str(), bl.length(), buf);
+  string buf = buf_to_hex(reinterpret_cast<unsigned char const * const>(bl.c_str()), bl.length());
```

Comment:
`unsigned char const * const` seems a bit pleonastic since casts return copies anyway.

Reply:
I don't believe there's any copy made here (or by most casts between non-class types). Anyway, it certainly does show some pedanticism-- in this case, certainly /excessive/, but not identical to the much more common but subtly incorrect <unsigned const char *> (in West-const), which I've also written plenty of. Maybe it was the coffee that morning. (And, I got to learn a new word, thanks!!)